src/camvidlog2/data.py

```
@@ -1,15 +1,49 @@
+    human_label: str = ""
+
+
+class FrameEmbedding(BaseModel):
+    source: Literal["frame"] = "frame"
+    filepath: Path
+    frame_no: int
+
+
+# see https://stackoverflow.com/a/70917353/932342
+AnyEmbedding = Annotated[
+    Union[StringEmbedding, FrameEmbedding], Field(discriminator="source")
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)